packages/bpk-component-navigation-bar/src/BpkNavigationBarIconButton.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React, { type ComponentType } from 'react';
+import PropTypes from 'prop-types';
+// TODO: close button is not really only a close button, we should rename and update the import here
+import BpkIconButton from 'bpk-component-close-button';
+import { withRtlSupport } from 'bpk-component-icon';
+
+export type Props = {
+  icon: ComponentType<any>,
+  label: string,
+  onClick: (event: SyntheticEvent<>) => void,
+  className: ?string,
+};
+
+const addRtlSupportIfNeeded = (component: ComponentType<any>) => {
```

Comment:
Can you explain whats happening here?

Reply:
Do you mean adding a comment or just explaining here :)? It's basically checking if the icon has been wrapped with rtl support already, to avoid doing so again. Re wrapping wouldn't be a problem though, I just though it'd be better to avoid it if we can.

Reply:
Ah! In that case I think we should remove it altogether - there are some icons that shouldn't be flipped. The rule of thumb is that if the icon conveys direction (i.e. an long-arrow-right) then flip it, if it doesn't (i.e.`chart`) then don't.

We should leave this up to userland :)